src/TestEnvironment.Docker.Containers.Mssql/MssqlContainerWaiter.cs

```diff
@@ -6,38 +6,31 @@
 
 namespace TestEnvironment.Docker.Containers.Mssql
 {
-    public class MssqlContainerWaiter : IContainerWaiter<MssqlContainer>
+    public class MssqlContainerWaiter : BaseContainerWaiter<MssqlContainer>
     {
-        private readonly ILogger _logger;
-
         public MssqlContainerWaiter(ILogger logger = null)
+            : base(logger)
         {
-            _logger = logger;
         }
 
-        public async Task<bool> Wait(MssqlContainer container, CancellationToken cancellationToken)
+        protected override async Task<bool> PerformCheckAsync(MssqlContainer container, CancellationToken cancellationToken)
         {
-            if (container == null) new ArgumentNullException(nameof(container));
-
             try
             {
-                using (var connection = new SqlConnection(container.GetConnectionString()))
-                using (var command = new SqlCommand("SELECT @@VERSION", connection))
-                {
-                    await connection.OpenAsync();
-                    await command.ExecuteNonQueryAsync();
-                }
+                using var connection = new SqlConnection(container.GetConnectionString());
+                using var command = new SqlCommand("SELECT @@VERSION", connection);
+                
+                await connection.OpenAsync(cancellationToken);
+                await command.ExecuteNonQueryAsync(cancellationToken);
 
                 return true;
             }
             catch (Exception ex) when (ex is InvalidOperationException || ex is NotSupportedException || ex is SqlException)
             {
-                _logger?.LogDebug(ex.Message);
+                Logger?.LogDebug(ex.Message);
```

Comment:
Probably we need to unify logging message in case of exception - see MongoDB waiter for example and apply such pattern to others

Reply:
Extracted to `BaseContainerWaiter`